agent/grpc-external/services/resource/watch.go

```diff
@@ -110,6 +102,11 @@ func (s *Server) validateWatchListRequest(req *pbresource.WatchListRequest) (*re
 		return nil, err
 	}
 
+	// if no tenancy is passed defaults to wildcard
+	if req.Tenancy == nil {
+		req.Tenancy = getDefaultTenancy(reg.Scope)
+	}
+
```

Comment:
Is there a way we can extract this before the validate func? I wouldn't expect a validate to mutate the request.

Reply:
I actually think we should keep this in place instead of splitting it out in https://github.com/hashicorp/consul/pull/19467/commits/c93cede992ed7938662c6859789a15d9d2e395fa.

I think the `validateWatchListRequest` function is doing the right thing, but the ergonomics of the function name don't align with what we traditionally think of "validate" only being a read-only operation.

This is further shown by the split of the `Registration.Mutate()` and `Registration.Validate()` functions.  We're essentially combining both here.

I'd like to see us just rename the function across all resource service endpoints to `ensure<Op>RequestValid()` which gets rid of the "read-only" stigma and seems reasonable as in "do whatever is necessary to make the request valid".